force_wfmanager/ui/setup/mco/kpi_specification_view.py

```diff
@@ -0,0 +1,308 @@
+from traits.api import (
+    Bool, Event, Instance, List, Property, Unicode, cached_property,
+    on_trait_change, HasTraits, Button
+)
+from traitsui.api import (
+    Item, View, ListEditor, TableEditor, ObjectColumn,
+    VGroup, HGroup, UReadonly, ModelView
+)
+
+from force_bdss.api import KPISpecification, BaseMCOModel
+from force_wfmanager.utils.variable_names_registry import (
+    VariableNamesRegistry
+)
+
+
+class TableRow(HasTraits):
+    """A representation of a variable in the workflow. Instances of TableRow
+    are displayed in a table using the TableEditor."""
+
+    #: The variable's type
+    type = Unicode()
+
+    #: The variable's name
+    name = Unicode()
+
+
+class KPISpecificationModelView(ModelView):
+
+    # -------------------
+    # Required Attributes
+    # -------------------
+
+    #: KPI model
+    model = Instance(KPISpecification)
+
+    # ------------------
+    # Regular Attributes
+    # ------------------
+
+    #: Defines if the KPI is valid or not. Set by the function
+    #: map_verify_workflow in workflow_view.py
+    valid = Bool(True)
+
+    #: An error message for issues in this modelview. Updated by
+    #: :func:`verify_tree
+    #: <force_wfmanager.ui.setup.workflow_tree.WorkflowTree.verify_tree>`
+    error_message = Unicode()
+
+    #: Event to request a verification check on the workflow
+    #: Listens to: :attr:`model.name <model>` and :attr:`model.type <model>`
+    verify_workflow_event = Event
+
+    # ------------------
+    #     Properties
+    # ------------------
+
+    #: Human readable label for ModelView
+    label = Property(Instance(Unicode),
+                     depends_on='model:[name,objective]')
+
+    # ------------------
+    #     View
+    # ------------------
+
+    # The traits_view only displays possible options for
+    # model.name listed in kpi_names. However, it is possible
+    # to directly change model.name without updating kpi_names
+    traits_view = View(
+        UReadonly('model.name'),
```

Comment:
This`UReadonly` trait view item used to use an `EnumEditor` to display a list of possible KPI names. However, the new design of the `KPISpecificationView` moves the KPI name selection to before instantiation of the `KPISpecificationModelView`, so the ability to re-select this name feels redundant

Reply:
Agreed, though this might be best dealt with at the `WorkflowView` or `WorkflowTree` level, since ideally if the user updates a name in a data source this should be picked up by the MCO and the corresponding KPI/Parameter name synced. At the moment, the `KPISecificationView` only checks for trait names in lists, without any knowledge of what the traits actually are.